datastore/tests/integration/smoke_test.py

```diff
@@ -245,6 +246,41 @@ async def test_gql_query(creds: str, kind: str, project: str) -> None:
         assert len(after.entity_results) == num_results + 3
 
 
+@pytest.mark.asyncio
+@pytest.mark.xfail(strict=False)
```

Comment:
Could you please explain why you've marked this as expecting to fail? The test makes sense to me!

Reply:
The output of this test depends on the records being inserted to be consistent before the reads. Given the eventual consistency property of Google's Datastore, the test could fail, event using the datastore emulator. I was not able to make it pass 100% of the times, even trying things like `consistency.STRONG` on the `runQuery` calls.

Reply:
Im sure the same happened with the test above this one, which I used to based my test.

Reply:
That's too bad. Man, datastore is such a pain to work with haha. I'll never understand why the world starting thinking not having ACID-complaince in your database was a good idea...

Reply:
Indeed! Took me a while to understand what was happening in this simple test.